planemo/shed2tap/base.py

```
@@ -301,24 +357,181 @@ def from_elem(cls, elem, package):
+    """
+    answer = []
+    local = _cache_download(url, downloaded_filename)
+
+    if tarfile.is_tarfile(local):
+        folders = _tar_folders(local)
+        if downloaded_filename.endswith(".tar.gz") or downloaded_filename.endswith(".tgz"):
+            answer.append('tar -zxvf %s' % downloaded_filename)
+        elif downloaded_filename.endswith(".tar.bz2"):
+            answer.append('tar -jxvf %s' % downloaded_filename)
+        elif downloaded_filename.endswith(".tar"):
+            answer.extend('tar -xvf %s' % downloaded_filename)
+        elif downloaded_filename.endswith(".dmg"):
+            pass
+        else:
+            # Quite possibly this file doesn't need decompressing,
+            # but until we've tested lots of real world tool_dependencies.xml
+            # files I'd like to check these cases to confirm this.
+            raise NotImplementedError("How to decompress tar file %s?" % downloaded_filename)
+    elif zipfile.is_zipfile(local) and not local.endswith('.jar'):
+        folders = _zip_folders(local)
+        answer.append('unzip %s' % downloaded_filename)
+    else:
+        # No compression? Leave as it is?
+        raise NotImplementedError("What kind of compression is %s using?" % local)
+
+    common_prefix = _common_prefix(folders)
+    if common_prefix:
+        answer.append('cd "%s"' % common_prefix)
+
+    return answer
+
+
+def _commands_to_download_and_extract(url, target_filename=None):
+    # TODO - Include checksum validation here?
+    if target_filename:
+        downloaded_filename = target_filename
+    else:
+        downloaded_filename = os.path.split(url)[-1]
+        if "?" in downloaded_filename:
+            downloaded_filename = downloaded_filename[:downloaded_filename.index("?")]
+        if "#" in downloaded_filename:
+            downloaded_filename = downloaded_filename[:downloaded_filename.index("#")]
+
+    # Curl is present on Mac OS X, can we assume it will be on Linux?
```

Good - setting the download target filename with ``wget`` is tricky.

If curl is missing this would fail noisily and clearly at least :)